autorest/azure/rp.go

```diff
@@ -0,0 +1,195 @@
+package azure
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// RegisterResourceProvider  tries to register the resource provider in case it is unregistered
+func RegisterResourceProvider(client autorest.Client) autorest.SendDecorator {
```

Comment:
Since this is replacing DoRetryForStatusCodes() and augmenting it with registration I would rename this to something that reflects that, maybe DoRetryWithRegistration() or something along those lines.

Reply:
Fixed